lib/src/source_visitor.dart

```diff
@@ -2599,6 +2623,53 @@ class SourceVisitor extends ThrowingAstVisitor {
     visit(node.operand);
   }
 
+  @override
+  void visitRepresentationConstructorName(RepresentationConstructorName node) {
+    token(node.period);
+    token(node.name);
+  }
+
+  @override
+  void visitRepresentationDeclaration(RepresentationDeclaration node) {
+    // Consider having `node.asFormalParameterList()`, and format based on that.
```

Comment:
[nit] I don't think this is a useful comment unless it's a TODO with an issue attached.